core/src/test/scala/org/apache/spark/deploy/ExternalShuffleServiceDbSuite.scala

```diff
@@ -0,0 +1,193 @@
+    dataContext.create()
+    // Write some sort data.
+    dataContext.insertSortShuffleData(0, 0,
+      Array[Array[Byte]](sortBlock0.getBytes(StandardCharsets.UTF_8),
+        sortBlock1.getBytes(StandardCharsets.UTF_8)))
+    registerExecutor()
+  }
+
+  override def afterAll() {
+    try {
+      dataContext.cleanup()
+    } finally {
+      super.afterAll()
+    }
+  }
+
+  def registerExecutor(): Unit = {
+    sparkConf.set("spark.shuffle.service.db.enabled", "true")
+    externalShuffleService = new ExternalShuffleService(sparkConf, securityManager)
+
+    // external Shuffle Service start
+    externalShuffleService.start()
+    bockHandler = externalShuffleService.getBlockHandler
+    blockResolver = bockHandler.getBlockResolver
+    blockResolver.registerExecutor("app0", "exec0", dataContext.createExecutorInfo(SORT_MANAGER))
+    blockResolver.closeForTest()
+    // external Shuffle Service stop
+    externalShuffleService.stop()
+  }
+
+  // This test getBlockData will be passed when the external shuffle service is restarted.
+  test("restart External Shuffle Service With InitRegisteredExecutorsDB") {
+    sparkConf.set("spark.shuffle.service.db.enabled", "true")
+    externalShuffleService = new ExternalShuffleService(sparkConf, securityManager)
+    // externalShuffleService restart
+    externalShuffleService.start()
+    bockHandler = externalShuffleService.getBlockHandler
+    blockResolver = bockHandler.getBlockResolver
+
+    val block0Stream = blockResolver.getBlockData("app0", "exec0", 0, 0, 0).createInputStream
+    val block0 = CharStreams.toString(new InputStreamReader(block0Stream, StandardCharsets.UTF_8))
+    block0Stream.close()
+    assert(sortBlock0 == block0)
+    // pass
+    blockResolver.closeForTest()
+    // externalShuffleService stop
+    externalShuffleService.stop()
+
+  }
+
+  // This test getBlockData will't be passed when the external shuffle service is restarted.
+  test("restart External Shuffle Service Without InitRegisteredExecutorsDB") {
+    sparkConf.set("spark.shuffle.service.db.enabled", "false")
+    externalShuffleService = new ExternalShuffleService(sparkConf, securityManager)
+    // externalShuffleService restart
+    externalShuffleService.start()
+    bockHandler = externalShuffleService.getBlockHandler
+    blockResolver = bockHandler.getBlockResolver
+
+    val error = intercept[RuntimeException] {
+      val block0Stream = blockResolver.getBlockData("app0", "exec0", 0, 0, 0).createInputStream
+      val block0 = CharStreams.toString(new InputStreamReader(block0Stream, StandardCharsets.UTF_8))
+      block0Stream.close()
+      assert(sortBlock0 == block0)
+    }.getMessage
+
+    assert(error.contains("not registered"))
+    blockResolver.closeForTest()
+    // externalShuffleService stop
+    externalShuffleService.stop()
+  }
+
+  /**
+   * Manages some sort-shuffle data, including the creation
+   * and cleanup of directories that can be read by the
```

Comment:
comment appears to be cutoff.

I don't love that this is copied from elsewhere -- can you just add the network-common as a test dependency to core, like this:

https://github.com/apache/spark/blob/827d3718770e2c65821a1f28b889743ba872b7cf/core/pom.xml#L364-L370